File: benchmarks/word_count/word_count.py
# Word count benchmark - Python
# Count words and lines in large generated text (classic MapReduce task)

import sys

n = int(sys.argv[1]) if len(sys.argv) > 1 else 100000

phrase = "the quick brown fox jumps over the lazy dog\n"

# Generate text: repeat phrase n times
text = phrase * n

# Count words and lines
words = len(text.split())
lines = text.count('\n')

print(f"{words} {lines}")

File: benchmarks/word_count/word_count.js
// Word count benchmark - JavaScript
// Count words and lines in large generated text (classic MapReduce task)

const n = parseInt(process.argv[2]) || 100000;

const phrase = "the quick brown fox jumps over the lazy dog\n";

// Generate text: repeat phrase n times
const text = phrase.repeat(n);

// Count words and lines
const words = text.split(/\s+/).filter(w => w.length > 0).length;
const lines = (text.match(/\n/g) || []).length;

console.log(`${words} ${lines}`);
